Tests/Unit/EmogrifierTest.php

```diff
@@ -670,6 +670,8 @@ public function selectorDataProvider()
                 => ['p:last-child {' . $styleRule . '} ', '#<p class="p-6" style="' . $styleRule . '">#'],
             'DIV:last-child does not match last child with mismatching tag'
                 => ['div:last-child {' . $styleRule . '} ', '#<p class="p-6">#'],
+            'P:not(.p-6) matches all but P.p-6'
+                => ['p:not(.p-6) {' . $styleRule . '}', '#<p class="p-6">#'],
```

Comment:
These should be two separate tests (i.e., cases in the data provider): One that it matches the other elements, and one that it does not match the corresponding element.